### PR TITLE
Add privacy config plugin

### PR DESCRIPTION
**Pull request by aleqsio**
# Why

Related to https://github.com/expo/expo/issues/27796
We want to allow users to define [Privacy Manifests](https://developer.apple.com/documentation/bundleresources/privacy_manifest_files/describing_data_use_in_privacy_manifests) in expo config.

# How

Added a new default prebuild config plugin.
Docs will follow up in a separate PR.

# Test Plan

Tested on native-component-list – the file gets added and included in the report.

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

**Comment by aleqsio** [comment]
I see the check-packages lint violations, will fix before merging

File: packages/@expo/config-types/src/ExpoConfig.ts [hunk 1]
```diff
@@ -404,6 +404,23 @@ export interface IOS {
   entitlements?: {
     [k: string]: any;
   };
+  /**
+   * Configuration to add to your standalone app's native *.xcprivacy file. [Learn more](https://developer.apple.com/documentation/bundleresources/privacy_manifest_files).
```
Comment (brentvatne) [inline packages/@expo/config-types/src/ExpoConfig.ts:408]:
```suggestion
   * Configuration to add to your app's native *.xcprivacy file. [Learn more](https://developer.apple.com/documentation/bundleresources/privacy_manifest_files).
```

**Review by brentvatne — APPROVED** [review]
we generally don't want to add new fields to the base expo config, but i think this is one of the cases where it's reasonable to make an exception to that

**Comment by aleqsio** [comment]
@brentvatne Agreed, but we don't have a natural place to put it in (considered making expo-privacy that is just that config plugin for now but seemed like a worse option)

**Review by tsapeta — APPROVED** [review]

**Pull request merged**